### DISCUSSION TITLE
How to let user crop Freely

### DISCUSSION THREAD

**Discussion by sad-adnan** [hm21/pro_image_editor#232] [Q&A] [answered]
i currently have below code to give users aspect ratio options, but the free ratio is not working. any idea how to achieve this

cropRotateEditorConfigs: const CropRotateEditorConfigs(
              enabled: true,
              canRotate: true,
              canChangeAspectRatio: true,
              initAspectRatio: 0.0,
              aspectRatios: [
                AspectRatioItem(text: 'Free', value: null),
                AspectRatioItem(text: 'Original', value: 0.0),
                AspectRatioItem(text: '1:1', value: 1.0 / 1.0),
                AspectRatioItem(text: '3:2', value: 3.0 / 2.0),
                AspectRatioItem(text: '2:3', value: 2.0 / 3.0),
                AspectRatioItem(text: '4:3', value: 4.0 / 3.0),
                AspectRatioItem(text: '3:4', value: 3.0 / 4.0),
                AspectRatioItem(text: '16:9', value: 16.0 / 9.0),
                AspectRatioItem(text: '9:16', value: 9.0 / 16.0),
                AspectRatioItem(text: '3:5.5', value: 3.0 / 5.5),
              ],
            ),

**Comment by hm21** [comment, marked as answer]
To enable a free aspect ratio, set the value to `-1`. Below are the default values for reference.



```dart
this.aspectRatios = const [
  AspectRatioItem(text: 'Free', value: -1),
  AspectRatioItem(text: 'Original', value: 0.0),
  AspectRatioItem(text: '1*1', value: 1.0 / 1.0),
  AspectRatioItem(text: '4*3', value: 4.0 / 3.0),
  AspectRatioItem(text: '3*4', value: 3.0 / 4.0),
  AspectRatioItem(text: '16*9', value: 16.0 / 9.0),
  AspectRatioItem(text: '9*16', value: 9.0 / 16.0)
],
```

**Reply by sad-adnan** [reply]
@hm21 Thanks a lot.
